### DISCUSSION TITLE
Setup function not callable if empty opts are passed

### DISCUSSION THREAD

**Discussion by salvaft** [stevearc/dressing.nvim#98] [Q&A] [answered]
I am using Lazy. Might be related to the plugin manager, i dont know.
If I install the plugin using according to the readme: 
```
{
  'stevearc/dressing.nvim',
  opts = {},
}

```
And later on in my config file I try to (for example):
```
require("dressing").setup({
	select = {
		get_config = function(opts)
			if opts.kind == "codeaction" then
				return {
					telescope = require("telescope.themes").get_cursor({ initial_mode = "normal" }),
				}
			end
		end,
	},
})
```
The setup call is doing nothing. To get it working I have to remove opts from here:
```
{
  'stevearc/dressing.nvim',
}
```
Then if do multiple setup calls, the only the last one is taken into account

**Comment by stevearc** [comment, marked as answer]
dressing.nvim will set its config options to be whatever the latest call to `setup` passes in
https://github.com/stevearc/dressing.nvim/blob/f19cbd56f7f8cad212c58a7285d09c5d9c273896/lua/dressing/init.lua#L5-L6

The issue is that you're fighting with lazy.nvim. It expects you to handle your setup in the definition of the plugin, either by using `opts = {...}`, or `config = true`, or 
```lua
config = function(plugin, opts)
  require('dressing').setup({})
end
```

Defining the plugin in one place and calling the `setup` function in another does work, but isn't really how it was meant to be used. And as you have discovered you will want to be careful about what options you pass around because setting `opts = {}` will cause lazy to call `require("dressing").setup()` which can conflict with your manual call elsewhere.
